Source/Events/EventType.cs

```diff
@@ -13,31 +13,64 @@ public class EventType : Artifact<EventTypeId>
         /// <summary>
         /// Initializes a new instance of the <see cref="EventType"/> class.
         /// </summary>
-        /// <param name="id">The <see cref="EventTypeId">unique identifer</see> of the <see cref="EventType"/>.</param>
+        /// <param name="id">The <see cref="EventTypeId">unique identifier</see> of the <see cref="EventType"/>.</param>
         public EventType(EventTypeId id)
+            : this(id, alias: null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventType"/> class.
+        /// </summary>
+        /// <param name="id">The <see cref="EventTypeId">unique identifier</see> of the <see cref="EventType"/>.</param>
+        /// <param name="alias"><see cref="EventTypeAlias">Alias</see> of the <see cref="EventType"/>.</param>
+        public EventType(EventTypeId id, EventTypeAlias alias)
             : base(id)
         {
             ThrowIfEventTypeIdIsNull(id);
+            Alias = alias;
         }
 
         /// <summary>
         /// Initializes a new instance of the <see cref="EventType"/> class.
         /// </summary>
-        /// <param name="id">The <see cref="EventTypeId">unique identifer</see> of the <see cref="EventType"/>.</param>
+        /// <param name="id">The <see cref="EventTypeId">unique identifier</see> of the <see cref="EventType"/>.</param>
         /// <param name="generation"><see cref="Generation">Generation</see> of the <see cref="EventType"/>.</param>
         public EventType(EventTypeId id, Generation generation)
+            : this(id, generation, null)
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="EventType"/> class.
+        /// </summary>
+        /// <param name="id">The <see cref="EventTypeId">unique identifier</see> of the <see cref="EventType"/>.</param>
+        /// <param name="generation"><see cref="Generation">Generation</see> of the <see cref="EventType"/>.</param>
+        /// <param name="alias"><see cref="EventTypeAlias">Alias</see> of the <see cref="EventType"/>.</param>
+        public EventType(EventTypeId id, Generation generation, EventTypeAlias alias)
             : base(id, generation)
         {
             ThrowIfEventTypeIdIsNull(id);
             ThrowIfGenerationIsNull(generation);
+            Alias = alias;
         }
 
-        void ThrowIfEventTypeIdIsNull(EventTypeId id)
+        /// <summary>
+        /// Gets the alias for the Event Type.
+        /// </summary>
+        public EventTypeAlias Alias { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether the Event Type has an alias or not.
+        /// </summary>
+        public bool HasAlias => Alias?.Value != default;
```

Comment:
Always makes me uneasy when we have to check nulls, is it not possible to set this boolean somehow from the constructors?